platform/darwin/src/MGLGeometryOfflineRegion.h

```diff
@@ -0,0 +1,80 @@
+#import <Foundation/Foundation.h>
+
+#import "MGLFoundation.h"
+#import "MGLOfflineRegion.h"
+#import "MGLShape.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/**
+ An offline region defined by a style URL, geographic coordinate bounds, and
+ range of zoom levels.
```

Comment:
Are there any benefits to using an MGLTilePyramidOfflineRegion instead of an MGLShapeOfflineRegion going forward, such as performance? If so, we can mention those benefits here and in MGLTilePyramidOfflineRegion’s documentation comment.

Reply:
`MGLTilePyramidOfflineRegion` would perform better at computing the number of tile resources needed, and computing their actual IDs. But that performance advantage over `MGLShapeOfflineRegion` should be marginal when including the download as well. 

More importantly, switching from using the bounding box to an actual shape would reduce the number of resources required, the storage and download cost. It would also help with the tile download ceiling for many customers interested in many small offline regions.

Reply:
In that case, let’s mention the new class’s benefits in the documentation comments for both classes. The old class is being left in place for backwards compatibility reasons, but we don’t have to say that just yet since we aren’t deprecating it.